### PR TITLE
Fine-Tuning Scheduler Tutorial Update for Lightning/PyTorch `2.4.0`

### PR DESCRIPTION
**Pull request by speediedan**
Validated (locally) the Fine-Tuning Scheduler (FTS) tutorial for FTS/Lightning/PyTorch `2.4.0` (as of the [recent `2.4.0` Lightning commit](https://github.com/Lightning-AI/lightning/tree/2064887b12dd934a5f9a2bf45897f29e3bfc74d1)). 
(this PR is currently using `2.3.3` until Lightning and FTS `2.4.x` are released)

The only minor changes in this PR are to explicitly set `datasets` `trust_remote_code` (which will be mandatory with Datasets 3.x) and to remove a reference to the now unsupported PT `2.0.x`

Thank you for all your work and your consistently valuable contributions to the open-source ML community!

## Before submitting

- [X] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [X] Did you make sure to **update the docs**?
- [X] Did you write any new **necessary tests**?

## What does this PR do?

Updates the Fine-Tuning Scheduler tutorial to prepare for `2.4.0` tutorial publishing.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: lightning_examples/finetuning-scheduler/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,3 @@
-lightning  # todo: the tuner depends on L so later using with PL crash
-finetuning-scheduler[examples] ==2.3.*
-datasets ==2.17.*
-# todo: pin version intill reinstall with PT-eco alignment
-torch ==2.1.*
-torchvision ==0.16.*
+datasets >=2.17.0  # to allow explicitly setting `trust_remote_code`
+lightning <=2.4.0  # todo: the tuner depends on L so later using with PL crash
```
Comment (Borda) [inline lightning_examples/finetuning-scheduler/requirements.txt:2]:
```suggestion
lightning <2.4.0  # todo: the tuner depends on L so later using with PL crash
```
also, would it be possible to make it work also with PL, let's see for example
https://github.com/Lightning-AI/lightning-Habana/blob/2fd33c6512155a1504df7d13b637ba3b0edaff8e/src/lightning_habana/utils/hpu_distributed.py#L19-L26

Comment (Borda) [reply lightning_examples/finetuning-scheduler/requirements.txt:2]: also, would it be possible to make it work also with PL, let's see for example
https://github.com/Lightning-AI/lightning-Habana/blob/2fd33c6512155a1504df7d13b637ba3b0edaff8e/src/lightning_habana/utils/hpu_distributed.py#L19-L26

**Review by Borda — DISMISSED** [review]

File: lightning_examples/finetuning-scheduler/requirements.txt (hunk 1 above)
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/requirements.txt]: I've tested locally with the `2.4.0` stack of lightning, pytorch and FTS and was thinking we could hold merging this until the Lightning and FTS official `2.4.x` releases are issued if you want to (either way, I'll re-run the tutorial when Lightning and FTS `2.4.0`  are out to verify everything looks good).

Comment (speediedan) [reply lightning_examples/finetuning-scheduler/requirements.txt]: I'll take a look at making the example L vs PL agnostic as you suggest and push a commit, should be possible (FTS can be used with either installation mode)

Comment (Borda) [reply lightning_examples/finetuning-scheduler/requirements.txt]: The reason is that it moved all L to PL, but then it complains about inheritance even though the code is identical. Regarding the 2.4, it is great it is compatible but until it is tested I would leave it with the latest released, and now the dependable we will have bumps tested as you can see for example #329 #345 #347

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 2]
```diff
@@ -172,13 +172,22 @@
...
+if module_available("lightning"):
+    import lightning as L
+    from lightning.pytorch.callbacks import EarlyStopping, ModelCheckpoint
+    from lightning.pytorch.loggers.tensorboard import TensorBoardLogger
+    from lightning.pytorch.utilities import rank_zero_warn
+elif module_available("pytorch_lightning"):
+    import pytorch_lightning as L
+    from pytorch_lightning.callbacks import EarlyStopping, ModelCheckpoint
+    from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
+    from pytorch_lightning.utilities import rank_zero_warn
```
Comment (Borda) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:186]:
```suggestion
```
it won't work here until it is the same also `finetuning_scheduler`
you can test it by removing `lightning` from the requirements

Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py]: reverted to previous Lightning import approach as discussed and all is passing. 

FTS will install Lightning by default so we don't need to include the `lightning` dependency in the notebook-specific dependencies. 

Note if [one installs finetuning-scheduler](https://github.com/speediedan/finetuning-scheduler/blob/04350f123ef9f72a9ccdd90e9120803f01469742/README.md?plain=1#L81) with `export PACKAGE_NAME=pytorch` though it will install using the standalone `pytorch-lightning` dependency and the example should work with `pytorch-lightning` as well (when changing the imports manually or introspecting them as was done in a previous commit). 

Probably makes sense to keep the example as is for now since anyone installing FTS with that non-default mode will know the example imports will need to change and we'd prefer to keep the example as simple as possible. Thanks again!

**Review by Borda — APPROVED** [review]
